### ISSUE TITLE
Questions on how to train the network

### ISSUE THREAD

**Issue by catherineytw** [soshishimada/Neural_Physcap_Demo#2]
Hi soshi,

Thank you for releasing the code. Really nice work！

I've been reading your paper and studying your code lately, and am curious about how to train the network. Here are my questions:
1.  What are the datasets you use to train the TPNet and GRFNet respectively? If I want to retrain the network, how to prepare the training and evaluation data?
2. Are TPNet and GRFNet trained jointly or separately? Is it possible to release the network training code and give us some training guidance?
3.  I am also curious about the  Lirr loss, which keeps the estimated joint angles in a reasonable range. As shown in the equation, I am wondering how to define the reasonable angle range for each joint?  From you paper and code, I infered that the 40-d angle vector was defined as (16-6)*4, could you provide more details on the definition? It was reasonable to exclude the estimation of the 4 end-effectors' angles, but I am curious about the other 2 joints you exlcuded. 
![image](https://user-images.githubusercontent.com/13126608/167532731-4537c66c-bf86-4cba-ac1b-bdc48224eced.png)

Thank you! Any responses will be highly appreciated!

Best regards!

**Comment by soshishimada** [comment]
> Hi soshi,
> 
> Thank you for releasing the code. Really nice work！
> 
> I've been reading your paper and studying your code lately, and am curious about how to train the network. Here are my questions:
> 
> 1. What are the datasets you use to train the TPNet and GRFNet respectively? If I want to retrain the network, how to prepare the training and evaluation data?
> 2. Are TPNet and GRFNet trained jointly or separately? Is it possible to release the network training code and give us some training guidance?
> 3. I am also curious about the  Lirr loss, which keeps the estimated joint angles in a reasonable range. As shown in the equation, I am wondering how to define the reasonable angle range for each joint?  From you paper and code, I infered that the 40-d angle vector was defined as (16-6)*4, could you provide more details on the definition? It was reasonable to exclude the estimation of the 4 end-effectors' angles, but I am curious about the other 2 joints you exlcuded.
>    ![image](https://user-images.githubusercontent.com/13126608/167532731-4537c66c-bf86-4cba-ac1b-bdc48224eced.png)
> 
> Thank you! Any responses will be highly appreciated!
> 
> Best regards!

1) TPNet and GRFNet are trained on Human3.6M, MPI-3DHP and DeepCap datasets. For TPNet, you'll need I) 2D joint keypoints, ii) known camera intrinsics and iii)corresponding ground truth "q" that consists of root translation, rotation and joint angles.  You don't need ground truth forces for training GRFNet since the pseudo GT force is obtained from the equation (7).
2) We first trained the networks separately. Then we trained all the networks jointly for a few epochs. Training script (or instruction) will be released soon.
3) We applied Lirr for all the joint angles (excluding end effectors). I guess correctly, it has to be "39"(=3*13), not "40" for the provided urdf file. I'll recheck and update the arxiv paper if necessary. The range is manually defined for each degree of freedom.
